src/app/app.component.ts

```diff
@@ -39,7 +45,18 @@ export class AppComponent implements OnInit {
 
   @HostListener('window:keydown', ['$event'])
   onKeypress(event: KeyboardEvent) {
-    const direction = getDirection(event.key);
+    const direction = this.directionHelper.getDirection(event.key);
+
+    if (direction === this.snake.movementDirection) {
```

Comment:
In practice, this shouldn't matter much but I think the comparison should be against the direction that the snake is facing.